packages/core/src/exchange/listToken.ts

```diff
@@ -69,11 +69,26 @@ export type Listing = {
    * @type Date
    */
   expiredAt?: Date
+
+  /**
+   * The metadata associated to the bid
+   * @type unknown
+   */
+  metadata?: unknown
```

Comment:
I wonder if we shouldn't restrict this a bit more with a record as anyway we cannot just put 42 otherwise it will fail

```suggestion
  metadata?: Record<string, unknown>
```

Reply:
I tested and the database accept `42` as well as array `[42, [42]]`. I would keep it as unrestricted as much as possible to allow any kind of data.
Also adding typescript type doesn't actually prevent any type of data to be passed at runtime.

Reply:
Don't you think it would be great to force the "key/value" structure as we already do in the dashboard also, to avoid breaking APIs, I would actually prefer to restrict as much as possible and then open if we can. 
Also, we did consider the metadata as key/value already in the admin and similar in the service you created

Reply:
That's a good point to make sure the metadata can be compatible-ish with the admin with a nice UI.
I will put the type `Record<string, unknown>` and create an issue for adding check in the database.

Reply:
Type changed and issue created: https://github.com/liteflow-labs/backend/issues/new